src/web/mjs/connectors/Novelgo.mjs

```diff
@@ -0,0 +1,93 @@
+        this.novelObstaclesQuery = 'div.code-block';
+        this.novelFormat = 'image/png';
+        this.novelWidth = '56em'; // parseInt(1200 / window.devicePixelRatio) + 'px';
+        this.novelPadding = '1.5em';
+    }
+
+    async _getMangasFromPage(page) {
+        const URI = new URL(this.path + page + '/', this.url);
+        let request = new Request(URI, this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryMangas);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, request.url),
+                title: element.text.trim()
+            };
+        });
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+        for(let page = 1, run = true; run; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    async _getChaptersFromJSON(page, id){
+        const URI = new URL('/wp-json/noveils/v1/chapters?paged=' + page + '&perpage=250&category=' + id, this.url);
+        let request = new Request(URI, this.requestOptions);
+        let data = await this.fetchJSON(request);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element.permalink, request.url),
+                title: element.post_title.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        const MANGA_ID = manga.id.substring(7).slice(0, -1);
+        let chapterList = [];
+        for (let page = 1, run = true; run; page++){
+            let chapters = await this._getChaptersFromJSON(page, MANGA_ID);
+            chapters.length > 0 ? chapterList.push(...chapters) : run = false;
+        }
+        return chapterList;
+    }
+
+    async _getPagesNovel(request) {
+        let script = `
+            new Promise(resolve => {
+                document.body.style.width = '${this.novelWidth}';
+                let container = document.querySelector('div.noveils-chapter-page div.container');
+                container.style.maxWidth = '${this.novelWidth}';
+                container.style.padding = '0';
+                container.style.margin = '0';
+                let novel = document.querySelector('div#chapter-post-content');
+                novel.style.padding = '${this.novelPadding}';
+                document.querySelectorAll('${this.novelObstaclesQuery}').forEach(element => element.style.display = 'none');
+                let script = document.createElement('script');
+                script.onload = async function() {
+                    let canvas = await html2canvas(novel);
+                    resolve(canvas.toDataURL('${this.novelFormat}'));
+                }
+                script.src = 'https://html2canvas.hertzen.com/dist/html2canvas.min.js';
```

Comment:
Unsure what's HN policy about external scripts inclusion (could be a security issue).
Need second point of view.

Reply:
`_getPagesNovel` function in this connector is same as `_getPagesNovel` function [WordpressMadaraNovel ](https://github.com/manga-download/hakuneko/blob/master/src/web/mjs/connectors/templates/WordPressMadaraNovel.mjs)that already being use in HakuNeko

Reply:
There is no problem as long as the script is included in the browser window that is opened with `fetchUI` in an isolated scope